main.nf

```diff
@@ -0,0 +1,932 @@
+params.thousandg = params.metaFiles[ params.genome ] ? params.metaFiles[ params.genome ].thousandg ?: false : false
+params.mills = params.metaFiles[ params.genome ] ? params.metaFiles[ params.genome ].mills ?: false : false
+params.omni = params.metaFiles[ params.genome ] ? params.metaFiles[ params.genome ].omni ?: false : false
+params.gfasta = params.metaFiles[ params.genome ] ? params.metaFiles[ params.genome ].gfasta ?: false : false
+params.bwa_index = params.metaFiles[ params.genome ] ? params.metaFiles[ params.genome ].bwa_index ?: false : false
+params.multiqc_config = "$baseDir/conf/multiqc_config.yaml"
+
+
+//Output configuration
+params.outdir = './results'
+params.saveAlignedIntermediates = false
+
+//Configuration parameters
+params.clusterOptions = false
+params.project = false
+params.cpus = 2
+
+
+// Show help when needed
+if (params.help){
+    log.info helpMessage
+    exit 0
+}
+
+// Nextflow version check
+nf_required_version = '0.25.0'
+try {
+    if( ! workflow.nextflow.version.matches(">= $nf_required_version") ){
+        throw GroovyException('Nextflow version too old')
+        }
+} catch (all) {
+    log.error "====================================================\n" +
+              "  Nextflow version $nf_required_version required! You are running v$workflow.nextflow.version.\n" +
+              "  Pipeline execution will continue, but things may break.\n" +
+              "  Please run `nextflow self-update` to update Nextflow.\n" +
+              "============================================================"
+}
+
+// Check blocks for ceratin required parameters, to see they are given and exists
+if (!params.reads || !params.genome){
+    exit 1, "Parameters '--reads' and '--genome' are required to run the pipeline"
+}
+if (!params.kitFiles[ params.kit ] && ['bait', 'target'].count{ params[it] } != 2){
+    exit 1, "Kit '${params.kit}' is not available in pre-defined config, so " +
+            "provide all kit specific files with option '--bait' and '--target'"
+}
+if (!params.metaFiles[ params.genome ] && ['gfasta', 'bwa_index', 'dbsnp', 'thousandg', 'clinvar', 'exac', 'gnomad'].count{ params[it] } != 7){
+    exit 1, "Genome '${params.genome}' is not available in pre-defined config, so you need to provide all genome specific " +
+            "files with options '--gfasta', '--bwa_index', '--dbsnp', '--thousandg', '--clinvar', '--exac' and '--gnomad'"
+}
+
+/*
+ * Create a channel for input read files
+ */
+Channel
+    .fromFilePairs( params.reads, size: params.singleEnd ? 1 : 2 )
+    .ifEmpty { exit 1, "Cannot find any reads matching: ${params.reads}\nNB: Path needs to be enclosed in quotes!\nNB: Path requires at least one * wildcard!\nIf this is single-end data, please specify --singleEnd on the command line." }
+    .into { read_files_fastqc; read_files_trimming }
+
+/**
+Validate Input indices for BWA Mem and GATK
```

Comment:
Inconsistent comment styles 😱 😆

Reply:
Was fixed in restructuring of input params commit.